Source/Diagnostics/ReducedDiags/FieldEnergy.cpp

```diff
@@ -0,0 +1,93 @@
+#include "FieldEnergy.H"
+#include "WarpX.H"
+#include "WarpXConst.H"
+#include "AMReX_REAL.H"
+#include "AMReX_ParticleReduce.H"
+#include <iostream>
+#include <cmath>
+
+using namespace amrex;
+
+/// constructor
+FieldEnergy::FieldEnergy (std::string rd_name)
+: ReducedDiags{rd_name}
+{
+
+    if (ParallelDescriptor::IOProcessor())
+    {
+        if ( m_IsNotRestart )
+        {
+            /// open file
+            std::ofstream ofs;
+            ofs.open(m_path + m_rd_name + ".txt",
+                std::ofstream::out | std::ofstream::app);
+            // write header row
+            ofs << "#";
+            ofs << "step";
+            ofs << m_sep;
+            ofs << "time(s)";
+            ofs << m_sep;
+            ofs << "total(J)";
+            ofs << m_sep;
+            ofs << "E(J)";
+            ofs << m_sep;
+            ofs << "B(J)";
+            ofs << std::endl;
+            /// close file
+            ofs.close();
+        }
+    }
+
+}
+///< end constructor
+
+/// function that computes field energy
+void FieldEnergy::ComputeDiags (int step)
+{
+
+    /// Judge if the diags should be done
+    if ( (step+1) % m_freq != 0 ) { return; }
+
+    /// get WarpX class object
+    auto & warpx = WarpX::GetInstance();
+
+    /// resize data array
+    /// the extra one is for total energy
+    m_data.resize(3,0.0);
+
+    /// get MultiFab data at level 0
+    const MultiFab & Ex = warpx.getEfield(0,0);
+    const MultiFab & Ey = warpx.getEfield(0,1);
+    const MultiFab & Ez = warpx.getEfield(0,2);
+    const MultiFab & Bx = warpx.getBfield(0,0);
+    const MultiFab & By = warpx.getBfield(0,1);
+    const MultiFab & Bz = warpx.getBfield(0,2);
```

Comment:
It looks like this would not work with mesh refinement. If this is true, could you put an error message (e.g., using `AMREX_ALWAYS_ASSERT_WITH_MESSAGE` in the WarpX constructor) of someone uses MR + reduced diagnostics? You can check that the error message is working with your input file.